integ/persistence_loading_policy_test.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/indeedeng/iwf/service/common/ptr"
+	"github.com/stretchr/testify/assert"
+	"strconv"
+	"testing"
+	"time"
+)
+
+func TestPersistenceLoadingPolicy_ALL(t *testing.T) {
+	for _, backendType := range getBackendTypes() {
+		for i := 0; i < *repeatIntegTest; i++ {
+			doTestPersistenceLoadingPolicy(t, backendType, persistence_loading_policy.LoadingTypeAll)
+			smallWaitForFastTest()
+		}
+	}
+}
+
+func TestPersistenceLoadingPolicy_PARTIAL_WITHOUT_LOCK(t *testing.T) {
+	for _, backendType := range getBackendTypes() {
+		for i := 0; i < *repeatIntegTest; i++ {
+			doTestPersistenceLoadingPolicy(t, backendType, persistence_loading_policy.LoadingTypePartialWithoutLock)
+			smallWaitForFastTest()
+		}
+	}
+}
+
+func TestPersistenceLoadingPolicy_PARTIAL_WITH_LOCK(t *testing.T) {
+	for _, backendType := range getBackendTypes() {
+		for i := 0; i < *repeatIntegTest; i++ {
+			doTestPersistenceLoadingPolicy(t, backendType, persistence_loading_policy.LoadingTypePartialWithLock)
+			smallWaitForFastTest()
+		}
+	}
+}
+
+func TestPersistenceLoadingPolicyTemporal_NONE(t *testing.T) {
+	for _, backendType := range getBackendTypes() {
+		for i := 0; i < *repeatIntegTest; i++ {
+			doTestPersistenceLoadingPolicy(t, backendType, persistence_loading_policy.LoadingTypeNone)
+			smallWaitForFastTest()
+		}
+	}
+}
+
+func doTestPersistenceLoadingPolicy(t *testing.T, backendType service.BackendType, loadingType string) {
+	wfHandler := persistence_loading_policy.NewHandler()
+	closeFunc1 := startWorkflowWorker(wfHandler)
+	defer closeFunc1()
+	closeFunc2 := startIwfService(backendType)
+	defer closeFunc2()
+
+	apiClient := iwfidl.NewAPIClient(&iwfidl.Configuration{
+		Servers: []iwfidl.ServerConfiguration{
+			{
+				URL: "http://localhost:" + testIwfServerPort,
+			},
+		},
+	})
+
+	wfId := persistence_loading_policy.WorkflowType + "_" + loadingType + "_" + strconv.Itoa(int(time.Now().UnixNano()))
+
+	wfInput := &iwfidl.EncodedObject{
+		Encoding: iwfidl.PtrString("json"),
+		Data:     iwfidl.PtrString(loadingType),
+	}
+
+	req := apiClient.DefaultApi.ApiV1WorkflowStartPost(context.Background())
+
+	startReq := iwfidl.WorkflowStartRequest{
+		WorkflowId:             wfId,
+		IwfWorkflowType:        persistence_loading_policy.WorkflowType,
+		WorkflowTimeoutSeconds: 10,
+		IwfWorkerUrl:           "http://localhost:" + testWorkflowServerPort,
+		StartStateId:           ptr.Any(persistence_loading_policy.State1),
+		StateInput:             wfInput,
+	}
+
+	_, httpResp, err := req.WorkflowStartRequest(startReq).Execute()
+	panicAtHttpError(err, httpResp)
+
+	time.Sleep(time.Second * 1)
+
+	history, data := wfHandler.GetTestResult()
+
+	assertions := assert.New(t)
+
+	assertions.Equalf(map[string]int64{
+		"S1_start":  1,
+		"S1_decide": 1,
+		"S2_start":  1,
+		"S2_decide": 1,
+	}, history, "persistence loading policy test fail, %v", history)
+
+	if loadingType == persistence_loading_policy.LoadingTypeAll {
+		assertions.Equalf(map[string]interface{}{
```

Comment:
Using the `data` for assertion is fine, but I just want to let you know that there is another way -- you can just do some assertion in the workflow code directly, and panic if not matched, like this one: https://github.com/indeedeng/iwf/blob/dbf26303ecfa3cc8a3dac0ad9f2759a892d98fff/integ/workflow/rpc/routers.go#L232

Initially I always use `data` to assert in test code, but later on I found that putting the assertion in workflow code(routers.go) is simpler to write the test and easier to read as well. :D

Reply:
Just tried it and found this way is really convenient!